### PR TITLE
Overextended functions

### PR DESCRIPTION
**Pull request by legende11**
## Description

It changes two of the qb-core functions to the ox_lib version. 

## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality. (couldn't test all of it)
- [X] My code fits the style guidelines.
- [X] My PR fits the contribution guidelines.

### CODE REVIEW

File: client/functions.lua [hunk 1]
```diff
@@ -71,10 +71,7 @@ end
 
 function QBCore.Functions.RequestAnimDict(animDict)
 	if HasAnimDictLoaded(animDict) then return end
```
Comment (FjamZoo) [inline client/functions.lua:73]: Might aswell remove this check as ox_lib function has the same check and its just duplicated code at this point.

We only convert these to OX for compatibility and people should just use ox functions natively anyway

File: client/functions.lua [hunk 2]
```diff
@@ -87,18 +84,12 @@ end
 
 function QBCore.Functions.LoadModel(model)
     if HasModelLoaded(model) then return end
```
Comment (FjamZoo) [inline client/functions.lua:86]: Same with this one

File: client/functions.lua [hunk 3]
```diff
@@ -87,18 +84,12 @@ end
...
-	RequestModel(model)
-	while not HasModelLoaded(model) do
-		Wait(0)
-	end
+    lib.requestModel(model, 100)
 end
 
 function QBCore.Functions.LoadAnimSet(animSet)
     if HasAnimSetLoaded(animSet) then return end
```
Comment (FjamZoo) [inline client/functions.lua:91]: And same with this one

File: client/functions.lua [hunk 4]
```diff
@@ -314,24 +295,14 @@ end
...
 function QBCore.Functions.GetClosestVehicle(coords)
     local ped = PlayerPedId()
-    local vehicles = GetGamePool('CVehicle')
-    local closestDistance = -1
-    local closestVehicle = -1
     if coords then
         coords = type(coords) == 'table' and vec3(coords.x, coords.y, coords.z) or coords
     else
         coords = GetEntityCoords(ped)
     end
```
Comment (FjamZoo) [inline client/functions.lua:302]: Here you can remove ped = PlayerPedId()

And where it uses ped use cache.ped

**Review by FjamZoo — CHANGES_REQUESTED** [review]
Left some changes that needs to be happen before this can be approved.

File: client/functions.lua [hunk 5]
```diff
@@ -71,10 +71,7 @@ end
 
 function QBCore.Functions.RequestAnimDict(animDict)
 	if HasAnimDictLoaded(animDict) then return end
-	RequestAnimDict(animDict)
-	while not HasAnimDictLoaded(animDict) do
-		Wait(0)
-	end
+	lib.requestAnimDict(animDict, 100)
 end
```
Comment (BerkieBb) [inline client/functions.lua:75]:
```suggestion
QBCore.Functions.RequestAnimDict = lib.requestAnimDict
```

File: client/functions.lua [hunk 6]
```diff
@@ -87,18 +84,12 @@ end
 
 function QBCore.Functions.LoadModel(model)
     if HasModelLoaded(model) then return end
-	RequestModel(model)
-	while not HasModelLoaded(model) do
-		Wait(0)
-	end
+    lib.requestModel(model, 100)
 end
```
Comment (BerkieBb) [inline client/functions.lua:88]:
```suggestion
QBCore.Functions.LoadModel = lib.requestModel
```

File: client/functions.lua [hunk 7]
```diff
@@ -87,18 +84,12 @@ end
...
 
 function QBCore.Functions.LoadAnimSet(animSet)
     if HasAnimSetLoaded(animSet) then return end
-    RequestAnimSet(animSet)
-    while not HasAnimSetLoaded(animSet) do
-        Wait(0)
-    end
+    lib.requestAnimSet(animSet, 100)
 end
```
Comment (BerkieBb) [inline client/functions.lua:93]:
```suggestion
QBCore.Functions.LoadAnimSet = lib.requestAnimSet
```

File: client/functions.lua [hunk 8]
```diff
@@ -274,21 +265,11 @@ function QBCore.Functions.GetClosestPlayer(coords)
...
-
-            if closestDistance == -1 or closestDistance > distance then
-                closestPlayer = closestPlayers[i]
-                closestDistance = distance
-            end
-        end
-    end
-    return closestPlayer, closestDistance
+
+    local playerId, playerPed, playerCoords = lib.getClosestPlayer(coords, 50, false)
```
Comment (BerkieBb) [inline client/functions.lua:269]: Replace `playerId` with an underscore here as it's not used

File: client/functions.lua [hunk 9]
```diff
@@ -314,24 +295,14 @@ end
...
-
-        if closestDistance == -1 or closestDistance > distance then
-            closestVehicle = vehicles[i]
-            closestDistance = distance
-        end
-    end
-    return closestVehicle, closestDistance
+    local Vehicle, VehicleCoords = lib.getClosestVehicle(coords, 50, true)
+    local closestDistance = #(VehicleCoords - coords)
+    return Vehicle, closestDistance
```
Comment (BerkieBb) [inline client/functions.lua:305]: Use camelCase for local variables

**Review by BerkieBb — CHANGES_REQUESTED** [review]

**Review by FjamZoo — APPROVED** [review]

**Review by BerkieBb — APPROVED** [review]

**Pull request merged**